`memcpy` operands overlap in `spid_buf`

Compiling with `gcc 13.2` at `-O3` has more aggressive in-lining causing detection of illegal `memcpy` operand overlap cases that were not found by earlier versions of `gcc`:

With tag `v1.0.2`
```
In function 'cyw43_send_ioctl',
    inlined from 'cyw43_do_ioctl' at lib/cyw43-driver/src/cyw43_ll.c:1184:15,
    inlined from 'cyw43_set_ioctl_u32' at lib/cyw43-driver/src/cyw43_ll.c:1822:5,
    inlined from 'cyw43_ll_wifi_on' at lib/cyw43-driver/src/cyw43_ll.c:1897:5:
lib/cyw43-driver/src/cyw43_ll.c:772:5: error: 'memcpy' accessing 4 bytes at offsets 68 and 68 overlaps 4 bytes at offset 68 [-Werror=restrict]
  772 |     memcpy(self->spid_buf + SDPCM_HEADER_LEN + 16, buf, len);
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
In this case `memcpy` source and destination coincide at identical offsets inside `spid_buf`.

Same error detected by `gcc 13.2` at several other locations in `cyw43_ll.c`.

Worked around problem by offsetting temporary operand locations in `spid_buf` by + 4. For example:
```c
static void cyw43_set_ioctl_u32(cyw43_int_t *self, uint32_t cmd, uint32_t val, uint32_t iface) {
    uint8_t *buf = &self->spid_buf[SDPCM_HEADER_LEN + 16 + 4];
    cyw43_put_le32(buf, val);
    cyw43_do_ioctl(self, SDPCM_SET, cmd, 4, buf, iface);
}
```

This quiets `gcc` error detection.

--- a/src/cyw43_ll.c
+++ b/src/cyw43_ll.c
@@ -746,7 +746,7 @@ static int cyw43_send_ioctl(cyw43_int_t *self, uint32_t kind, uint32_t cmd, size
     header->status = 0;
 
     // copy in payload
-    memcpy(self->spid_buf + SDPCM_HEADER_LEN + 16, buf, len);
+    memmove(self->spid_buf + SDPCM_HEADER_LEN + 16, buf, len);
 
     // do transfer
     CYW43_VDEBUG("Sending cmd %s (%u) len %u flags %u status %u\n", ioctl_cmd_name(header->cmd), header->cmd, header->len, header->flags, header->status);
@@ -783,7 +783,7 @@ int cyw43_ll_send_ethernet(cyw43_ll_t *self_in, int itf, size_t len, const void 
     if (is_pbuf) {
         pbuf_copy_partial((const struct pbuf *)buf, self->spid_buf + SDPCM_HEADER_LEN + 6, len, 0);
     } else {
-        memcpy(self->spid_buf + SDPCM_HEADER_LEN + 6, buf, len);
+        memmove(self->spid_buf + SDPCM_HEADER_LEN + 6, buf, len);
     }
 
     // do transfer

--- a/tests/sdio/Makefile
+++ b/tests/sdio/Makefile
@@ -1,3 +1,6 @@
+# Test with high optimisation level.
+CFLAGS += -O3
+
 SRC += \
 	main.c \
 	tests/mock_sdio.c \

